### PR TITLE
Regression: federated OIDC login broken in R4/zed keystone container 

### PR DESCRIPTION
**Pull request by JuanPTM**
https://github.com/SovereignCloudStack/issues/issues/313

### CODE REVIEW

**Comment by artificial-intelligence** [comment]
LGTM, but let's wait for the CI

**Review by matfechner — APPROVED** [review]
LGTM

**Comment by berendt** [comment]
Always wait for the CI before approving.

**Comment by berendt** [comment]
recheck

**Comment by berendt** [comment]
This will break the deployments before Zed:

```
failed: [testbed-node-0.testbed.osism.xyz] (item=RegionOne) => {"ansible_loop_var": "item", "changed": true, "changed_when_result": "The conditional check '(keystone_bootstrap.stdout | from_json).changed' failed. The error was: Expecting value: line 1 column 1 (char 0)", "cmd": ["docker", "exec", "keystone", "kolla_keystone_bootstrap", "admin", "password", "admin", "admin", "https://api-int.testbed.osism.xyz:5000", "https://api-int.testbed.osism.xyz:5000", "https://api.testbed.osism.xyz:5000", "RegionOne"], "delta": "0:00:00.031294", "end": "2023-04-25 06:24:56.153189", "item": "RegionOne", "msg": "non-zero return code", "rc": 1, "start": "2023-04-25 06:24:56.121895", "stderr": "Error response from daemon: Container 6122ab21aacf984724845fb46827509413ecdf2e26af16475399dcd8a6d27b9d is restarting, wait until the container is running", "stderr_lines": ["Error response from daemon: Container 6122ab21aacf984724845fb46827509413ecdf2e26af16475399dcd8a6d27b9d is restarting, wait until the container is running"], "stdout": "", "stdout_lines": []}
```

File: environments/kolla/files/overlays/keystone/wsgi-keystone.conf [hunk 1]
```diff
@@ -54,6 +54,7 @@ LogLevel info
 {% endif %}
 
 {% if keystone_enable_federation_openid %}
+    OIDCXForwardedHeaders X-Forwarded-Proto
```
Comment (matfechner) [inline environments/kolla/files/overlays/keystone/wsgi-keystone.conf:57]: Maybe as an idea something like this:
{% if keystone_oidc_forward_header is defined %}
    OIDCXForwardedHeaders "{{ keystone_oidc_forward_header }}"
{% endif %}

and in the environment/kolla/configuration.yml
under  section keystone

keystone_oidc_forward_header: "X-Forwarded-Proto"

**Review by matfechner — CHANGES_REQUESTED** [review]

File: environments/kolla/files/overlays/keystone/wsgi-keystone.conf (hunk 1 above)
Comment (JuanPTM) [reply environments/kolla/files/overlays/keystone/wsgi-keystone.conf]: Changes done. Let's see the CI results.

File: environments/kolla/configuration.yml [hunk 2]
```diff
@@ -46,6 +46,7 @@ elasticsearch_curator_hard_retention_period_days: 1
 
 # keystone
 enable_keystone_federation: "yes"
+keystone_oidc_forward_header: "X-Forwarded-Proto"
```
Comment (matfechner) [inline environments/kolla/configuration.yml:49]: @JuanPTM  Perhaps we should comment it  with #  so for upgrade pathes it won't block and for deployments with newer openstack your able to switch on

File: environments/kolla/files/overlays/keystone/wsgi-keystone.conf [hunk 3]
```diff
@@ -54,7 +54,7 @@ LogLevel info
 {% endif %}
 
 {% if keystone_enable_federation_openid %}
-{% if keystone_oidc_forward_header is defined %}
+{% if openstack_version == 'zed' %}
```
Comment (matfechner) [inline environments/kolla/files/overlays/keystone/wsgi-keystone.conf:57]: In my opinion this is more sustainable 
{% if openstack_version != 'yoga' %}

**Review by matfechner — CHANGES_REQUESTED** [review]

**Review by matfechner — APPROVED** [review]
LGTM but we should recheck the CI

**Comment by JuanPTM** [comment]
I've squashed the commits, CI was green, but lets wait for another run.

**Pull request merged**